localstack-core/localstack/services/apigateway/next_gen/execute_api/router.py

```diff
@@ -72,14 +72,14 @@ def prepare_rest_api_invocation(
         return context, response
 
     def is_rest_api(self, api_id: str, stage: str):
-        return (api_id, stage) in self._global_store.active_deployments
+        return (api := self._global_store.active_deployments[api_id]) and stage in api
```

Comment:
I think this will raise a KeyError if the API does not exist 
```suggestion
        return stage in self._global_store.active_deployments.get(api_id, {})
```

Reply:
It shouldn't raise an error. Given that it was created as a defaultdict:
```python
active_deployments: dict[str, dict[str, str]] = CrossAccountAttribute(lambda: defaultdict(dict))
```
Both would return the same value. We could however simplify my code to:
```suggestion
        return stage in self._global_store.active_deployments[api_id]
```

Reply:
If you feel we shouldn't use a `defaultdict` in in this case, I am open to the conversation.

My reasoning for using one in the first place is the following
- It simplifies and improves resilience for  all accesses to the nested dict. `self._global_store.internal_deployments[api_id][deployment_id]` vs `self._global_store.internal_deployments.get(api_id, {}][deployment_id]`
- `KeyError` will still be raised if the deployment doesn't not exists.
- `active_deployment` should not be regarded as a source of truth with regard to the existence of an api. So having all apis defaulting to an empty dict should not have adverse effect. :thinking: 

Once again, happy to discuss this further if you have a different point of view :smile:

Reply:
Right yes, I had forgotten about this!

I think the `defaultdict` is fine, it simplifies our job not having to use `setdefault` when creating deployments/stages for the API id level of the dict.

I guess my point is very nit-picky and niche, but accessing the defaultdict with `[]` will create an entry in the dict, where `.get()` doesn't. This `internal_deployments` will be stored with persistence and cloud pods, so we will store empty dicts if the API doesn't exist, and if we want to list API with deployments, we will get those non-existing keys. 

I think using defaultdict to automatically create the first level in the provider is good, but using it this way here might be a bit tricky, because it doesn't raise `KeyError`, but it still creates an entry in the dict. 

Maybe a bit of a mix, something like `(api := self._global_store.active_deployments.get(api_id)) and stage in api` can also work, but using the brackets when we're not sure it exists and can be accessed outside the provider feels a bit weird to me. But like I said, this is really, really nit-picky 😄

Reply:
I think the defaultdict mostly simplifies this for us in the provider:
```python
# creating a new deployment
_global_store.internal_deployments.setdefault(api_id, {})[deployment_id] = deployment

# vs
_global_store.internal_deployments[api_id][deployment_id] = deployment
```

Reply:
I didn't think of that effect on the store. :thinking:. It is a very valid point. Looks like the `defaultdict` isn't the right solution here, after all. It would keep that "danger" in the future as well.